### PR TITLE
Migrate to /opt/fluent directory

### PR DESCRIPTION
**Pull request by kenhys**
Migrate to /opt/fluent directory, command and service

Most of /opt/td-agent should be also migrated
from /opt/td-agent to /opt/fluent.

Additionally, path of commands or license file also renamed.

NOTE: For making safety migration process, the following are not moved
yet:

* /var/log/td-agent
* /etc/td-agent
* /etc/logrotate.d

To keep compatibility, the following command is still available.
(It is a symbolic link to /opt/fluent/bin/fluentd, /opt/fluent/bin/fluent-gem)

* /opt/fluent/bin/td-agent
* /opt/fluent/bin/td-agent-gem

### CODE REVIEW

**Comment by kenhys** [comment]
for now, migrating `c:\opt\td-agent` is out of scope in this PR (another PR)

**Comment by kenhys** [comment]
rerun failed jobs.

**Comment by kenhys** [comment]
waiting CI for focal.

File: fluent-package/yum/fluent-package.spec.in [hunk 1]
```diff
@@ -167,25 +167,51 @@ fi
...
+  if [ $(grep '\-b /opt/td-agent/bin/ruby' -c %{_sysconfdir}/prelink.conf) -eq 1 ]; then
+    # Migrate from td-agent v4
+    echo "old-prelink detected, Update /etc/prelink.conf ..."
+    %{__sed} -i"" %{_sysconfdir}/prelink.conf -e "s,/td-agent/,/fluent/,"
   fi
 fi
 
 %postun
 %systemd_postun_with_restart @SERVICE_NAME@.service
+if [ ! -f /usr/sbin/@COMPAT_SERVICE_NAME@ ]; then
```
Comment (ashie) [inline fluent-package/yum/fluent-package.spec.in:199]: The condition seems inverted:
```suggestion
if [ -f /usr/sbin/@COMPAT_SERVICE_NAME@ ]; then
```

File: fluent-package/yum/fluent-package.spec.in [hunk 2]
```diff
@@ -167,25 +167,51 @@ fi
...
-  elif [ $(grep '\-b /opt/td-agent/bin/ruby' -c %{_sysconfdir}/prelink.conf) -eq 0 ]; then
+  elif [ $(grep '\-b /opt/fluent/bin/ruby' -c %{_sysconfdir}/prelink.conf) -eq 0 ]; then
     echo "prelink detected, but /etc/prelink.conf.d/ does't exist. Adding /etc/prelink.conf ..."
-    echo "-b /opt/td-agent/bin/ruby" >> /etc/prelink.conf
+    echo "-b /opt/fluent/bin/ruby" >> /etc/prelink.conf
+  fi
+  if [ $(grep '\-b /opt/td-agent/bin/ruby' -c %{_sysconfdir}/prelink.conf) -eq 1 ]; then
+    # Migrate from td-agent v4
+    echo "old-prelink detected, Update /etc/prelink.conf ..."
+    %{__sed} -i"" %{_sysconfdir}/prelink.conf -e "s,/td-agent/,/fluent/,"
```
Comment (ashie) [inline fluent-package/yum/fluent-package.spec.in:193]: We prefer removing prelink settings completely, it's disabled by default as of RHEL 7 and not recommended.

e.g.)
* https://pagure.io/fesco/issue/1183
* https://support.oracle.com/knowledge/Oracle%20Linux%20and%20Virtualization/2658080_1.html

**Review by ashie — CHANGES_REQUESTED** [review]

File: fluent-package/yum/fluent-package.spec.in (hunk 1 above)
Comment (kenhys) [reply fluent-package/yum/fluent-package.spec.in]: Thanks, I've fixed it.

File: fluent-package/yum/fluent-package.spec.in (hunk 2 above)
Comment (kenhys) [reply fluent-package/yum/fluent-package.spec.in]: It seems that this topic should be treated in another PR.

Comment (kenhys) [reply fluent-package/yum/fluent-package.spec.in]: I'll create another PR.

**Review by ashie — APPROVED** [review]

**Comment by ashie** [comment]
Thanks!

**Pull request merged**